activesupport/lib/active_support/isolated_execution_state.rb

```diff
@@ -18,9 +18,9 @@ def isolation_level=(level)
         end
 
         if level != isolation_level
-          clear
-          singleton_class.alias_method(:current, "current_#{level}")
-          singleton_class.send(:private, :current)
+          clear if @isolation_level
+          singleton_class.alias_method(:executor, "current_#{level}")
```

Comment:
What do we think of making this more of a:

```ruby
self.scope = case level
             when :thread; Thread
             when :fiber; Fiber
             end
```

and then

```ruby
def context
  scope.current
end
```

(partly because I'd rather continue avoiding "executor"; partly because if we were going to use executor I feel like it more fits 'the thing that drives execution' than 'an execution', i.e. `Thread` not `Thread.current`; partly because I like the idea of removing this singleton class manipulation anyway)

Reply:
Ah yeah, 👍  to `scope`, that's a much better name than `executor`.

Reply:
I pushed another commit to resolve this, but just to be clear, there are few terms now in play based on this feedback (i.e. a few class methods, public or private, on `IsolatedExecutionState`):

- `scope` returns the Thread or Fiber class
- `context` returns the currently running instance of Thread or Fiber, which is now being used throughout my ConnectionPool changes as a reference to the current Thread or Fiber
- `state` returns the `active_support_execution_state` hash stashed on the `context`

Reply:
Cool, thanks -- that matches my intention, but I'll let @byroot confirm he's happy with those names / API.